include/index.h

```diff
@@ -149,11 +150,21 @@ template <typename T, typename TagT = uint32_t, typename LabelT = uint32_t> clas
     DISKANN_DLLEXPORT size_t search_with_tags(const T *query, const uint64_t K, const uint32_t L, TagT *tags,
                                               float *distances, std::vector<T *> &res_vectors);
 
+    // Filter support search String interface for converting to integer
+    template <typename IndexType>
+    DISKANN_DLLEXPORT std::unique_ptr<ANNReturnCode> search_with_filters(const T *query,
```

Comment:
ANNReturnCode only contain enum member, you can remove std::unique_ptr

Reply:
Its for better memory management, If I dont give unique ptr, it ends up with stack smashing